### PR TITLE
Line arrows example

### PR DESCRIPTION
**Pull request by fredj**
Add an example than shows how to draw arrows to a linestring (end of each segment)

### CODE REVIEW

**Comment by fredj** [comment]
Thanks for any review

File: src/ol/geom/linestring.js [hunk 1]
```diff
@@ -108,6 +109,23 @@ ol.geom.LineString.prototype.closestPointXY =
...
+ * If the callback returns a truthy value the function returns that
+ * value immediately. Otherwise the function returns `false`.
+ *
+ * @param {function(ol.Coordinate, ol.Coordinate): T} callback Function
+ *     called for each segment.
+ * @return {T|boolean} Value.
+ * @template T
+ * @api
+ */
+ol.geom.LineString.prototype.forEachSegments = function(callback) {
```
Comment (elemoine) [inline src/ol/geom/linestring.js]: `forEachSegments` -> `forEachSegment`

**Comment by fredj** [comment]
Note: an icon is used for the arrow but it should be possible to use a `ol.style.RegularShape` when an `anchor` options will be available (see #3342)

File: src/ol/geom/linestring.js (hunk 1 above)
Comment (fredj) [inline src/ol/geom/linestring.js]: yes, will change

**Comment by fredj** [comment]
@elemoine @tsauerwein: any other comments?

**Comment by tsauerwein** [comment]
No more objections from my side.

**Comment by elemoine** [comment]
Looking again…

**Comment by elemoine** [comment]
It looks good to me.

I just note that the style function will generate a lot of garbage if there is a lot of lines and segments. I know that this is not a problem in that particular example, but it would be good to always show good practices to users.

Instead of `function(ol.Feature, number):Array.<ol.style.Style>` the style function signature could possibly be `function(ol.Feature, number, function(ol.style.Style))`. And the style function would call the function passed as a third argument for each style it wants to generate.

So with this your example's style function could look like this:

```js
var lineStyle = new ol.style.Style({
  stroke: new ol.style.Stroke({
    // …
  })
});

var arrowStyle = new ol.style.Style({
  image: new ol.style.Icon({
    // …
  })
});
  

function styleFunction(feature, resolution, callback) {
  callback(lineStyle);
  feature.getGeometry().forEachSegment(function(start, end) {
    var dx = end[0] - start[0];
    var dy = end[1] - start[1];
    var rotation = Math.atan2(dy, dx);
    arrowStyle.setGeometry(new ol.geom.Point(end));
    arrowStyle.getImageStyle().setRotation(-rotation);
    callback(arrowStyle);
  });
}
```

Note something to do in this PR, but as a separate PR possibly.

**Comment by fredj** [comment]
Thanks for the review

**Pull request merged**